src/fstab-generator/fstab-generator.c

```diff
@@ -1347,11 +1347,11 @@ static int add_mounts_from_creds(bool prefix_sysroot) {
                                 /* prefix_sysroot = */ prefix_sysroot,
                                 /* accept_root = */ true,
                                 /* use_swap_enabled = */ true);
-                if (r < 0 && ret >= 0)
+                if (r != 0 && ret >= 0)
                         ret = r;
         }
 
-        return ret;
+        return ret; /* Return positive if at least one entry found. */
```

Comment:
I don't think this is going to work. We return the first error, or the true/false state. But we need to return both in parallel, i.e. we should set `enable_remount_fs` even if an unrelated error occured. I think the true/false info needs to be set via an output param.

Reply:
I noticed that systemd-remount-fs.service should not enabled when fstab=no even if systemd.mount-extra= or credential specifies mount points, as add_mount() honors specified mount options. The change is also dropped.